code/game/objects/items/robot/robot_parts.dm

```diff
@@ -237,9 +240,10 @@
 				O.notify_ai(1)
 				if(forced_ai)
 					O.connected_ai = forced_ai
-			if(!lawsync && !M.hacked)
+			if(!lawsync)
 				O.lawupdate = 0
-				O.make_laws()
+				if(!M.hacked && !M.clockwork)

```

Comment:
Also quite important. We don't want to override the laws we just got and we don't want to sync.